core/src/test/java/org/opensearch/sql/common/utils/StringUtilsTest.java

```diff
@@ -0,0 +1,46 @@
+package org.opensearch.sql.common.utils;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.opensearch.sql.common.utils.StringUtils.unquoteText;
+
+import org.junit.jupiter.api.Test;
+
+class StringUtilsTest {
+  @Test
+  void unquoteTest() {
+    assertEquals("test", unquoteText("test"));
+    assertEquals("test", unquoteText("'test'"));
+    assertEquals("test", unquoteText("`test`"));
+
+    assertEquals("test'", unquoteText("'test'''"));
+    assertEquals("test\"", unquoteText("\"test\"\"\""));
+
+    assertEquals("te``st", unquoteText("'te``st'"));
+    assertEquals("te``st", unquoteText("\"te``st\""));
+    assertEquals("te``st", unquoteText("`te``st`"));
+
+    assertEquals("te'st", unquoteText("'te''st'"));
+    assertEquals("te''st", unquoteText("\"te''st\""));
+    assertEquals("te''st", unquoteText("`te''st`"));
+
+    assertEquals("te\"\"st", unquoteText("'te\"\"st'"));
+    assertEquals("te\"st", unquoteText("\"te\"\"st\""));
+    assertEquals("te\"\"st", unquoteText("`te\"\"st`"));
+
+    assertEquals("''", unquoteText("''''''"));
+    assertEquals("\"\"", unquoteText("\"\"\"\"\"\""));
+    assertEquals("````", unquoteText("``````"));
+
+    assertEquals("test'", unquoteText("'test''"));
+
+    assertEquals("", unquoteText(""));
+    assertEquals("'", unquoteText("'"));
+    assertEquals("`", unquoteText("`"));
+    assertEquals("\"", unquoteText("\""));
+
+    assertEquals("hello'", unquoteText("'hello''"));
+    assertEquals("don't", unquoteText("'don't'"));
```

Comment:
Add tests for non-double internal " and ` too

Reply:
Can you please be more specific? Is that not likes 38/39.

Reply:
What about something like
```
assertEquals("hello`", unquoteText("`hello``"));
assertEquals("don"t", unquoteText(""don"t""));
```